crates/ruma-macros/src/lib.rs

```diff
@@ -108,27 +215,194 @@ pub fn event_enum(input: TokenStream) -> TokenStream {
     tokens.into()
 }
 
-/// Generates an implementation of `ruma_events::EventContent`.
+/// Generates traits implementations and types for an event content.
+///
+/// # Trait implementations
+///
+/// This macro implements the following traits for the type on which it is applied:
+///
+/// * `{kind}EventContent`
+/// * `StaticEventContent`
+/// * `StaticStateEventContent`, for the `State` kind.
+///
+/// # Generated types
+///
+/// It also generates type aliases and modified clones depending on the kind of event. To generate
+/// the base name of those types, the macro simply removes `Content` from the name of the type,
+/// which means that to apply this macro to a type, its name must always end with `Content`. And for
+/// compatibility with the [`event_enum!`] macro, the name should actually end with `EventContent`.
+///
+/// Some kinds can generate a modified clone of the event content type. For instance, for an event
+/// content type named `FooEventContent`:
+///
+/// * `RedactedFooEventContent`: the redacted form of the event content, for the `MessageLike` and
+///   `StateKind`. It also generates the `RedactContent` implementation which applies the redaction
+///   algorithm according to the Matrix specification.
+///
+///   The generated type implements `Redacted{Kind}EventContent`, `StaticEventContent`, `Serialize`
+///   and `Deserialize`.
+///
+///   The generation only works if the type is a struct with named fields. To keep a field after
+///   redaction, the `#[ruma_event(skip_redaction)]` attribute can be applied to that field.
+///
+///   To skip the generation of this type and trait to implement a custom redaction, or because it
+///   is not a struct with named fields, the `#[ruma_event(custom_redacted)]` attribute can be used
+///   on the container. The `RedactedFooEventContent` type must still exist and implement the same
+///   traits, even if it is only a type alias, and the `RedactContent` trait must still be
+///   implemented for those kinds.
```

Comment:
It's this because of the enum macro? If yes, maybe note that it only applies within ruma, not when using the derive for custom events.